src/Microsoft.ML.Recommender/MatrixFactorizationPredictor.cs

```diff
@@ -52,12 +52,16 @@ private static VersionInfo GetVersionInfo()
         private const uint VersionNoMinCount = 0x00010002;
 
         private readonly IHost _host;
+
         ///<summary> The number of rows.</summary>
-        public readonly int NumberOfRows;
+        public int NumberOfRows { get; }
+
         ///<summary> The number of columns.</summary>
-        public readonly int NumberOfColumns;
+        public int NumberOfColumns { get; }
+
         ///<summary> The rank of the factor matrices.</summary>
-        public readonly int ApproximationRank;
```

Comment:
>ApproximationRank [](start = 28, length = 17)

I'm not sure this one is any way better than previous code. Now you need to initialize new variable.